netbox/circuits/graphql/types.py

```diff
@@ -15,48 +20,93 @@
 )
 
 
-class CircuitTerminationType(CustomFieldsMixin, TagsMixin, CabledObjectMixin, ObjectType):
+@strawberry_django.type(
+    models.Provider,
+    fields='__all__',
+    filters=ProviderFilter
+)
+class ProviderType(NetBoxObjectType, ContactsMixin):
 
-    class Meta:
-        model = models.CircuitTermination
-        fields = '__all__'
-        filterset_class = filtersets.CircuitTerminationFilterSet
+    @strawberry_django.field
+    def networks(self) -> List[Annotated["ProviderNetworkType", strawberry.lazy('circuits.graphql.types')]]:
+        return self.networks.all()
```

Comment:
Its going to ProviderNetworkType, this is ProviderType

Reply:
What I mean is `ProviderType.networks` seems to return itself, for instance, because of the decorator. It's not a problem, just a bit confusing at first glance.